dispersy.py

```diff
@@ -1128,19 +1166,21 @@ def check_member_and_global_time(unique, times, message):
 
             key = (message.authentication.member.database_id, message.distribution.global_time)
             if key in unique:
-                return DropMessage(message, "already processed message by member^global_time")
+                returnValue(DropMessage(message, "already processed message by member^global_time"))
 
             else:
                 unique.add(key)
 
                 if not message.authentication.member.database_id in times:
-                    times[message.authentication.member.database_id] = [global_time for global_time, in self._database.execute(u"SELECT global_time FROM sync WHERE community = ? AND member = ? AND meta_message = ?",
-                                                                                                                               (message.community.database_id, message.authentication.member.database_id, message.database_id))]
+                    sync_packets = yield self._database.stormdb.fetchall(u"SELECT global_time FROM sync WHERE community = ? AND member = ? AND meta_message = ?",
+                                                                                                                               (message.community.database_id, message.authentication.member.database_id, message.database_id))
+                    times[message.authentication.member.database_id] = [global_time for global_time, in sync_packets]
                     assert len(times[message.authentication.member.database_id]) <= message.distribution.history_size, [message.packet_id, message.distribution.history_size, times[message.authentication.member.database_id]]
                 tim = times[message.authentication.member.database_id]
 
-                if message.distribution.global_time in tim and self._is_duplicate_sync_message(message):
-                    return DropMessage(message, "duplicate message by member^global_time (3)")
+                is_duplicate_sync_message = yield self._is_duplicate_sync_message(message)
+                if message.distribution.global_time in tim and is_duplicate_sync_message:
```

Comment:
I guess this `tim` variable should be named `time`? Maybe rename it (although you didn't create it in the first place)?